packages/components/src/toggle-control/index.tsx

```diff
@@ -0,0 +1,83 @@
+/**
+ * External dependencies
+ */
+import type { ChangeEvent } from 'react';
+import { css } from '@emotion/react';
+
+/**
+ * WordPress dependencies
+ */
+import { useInstanceId } from '@wordpress/compose';
+
+/**
+ * Internal dependencies
+ */
+import FormToggle from '../form-toggle';
+import BaseControl from '../base-control';
+import type { WordPressComponentProps } from '../ui/context/wordpress-component';
+import type { ToggleControlProps } from './types';
+import { contextConnect } from '../ui/context';
+import { HStack } from '../h-stack';
+import { useCx } from '../utils';
+import { space } from '../ui/utils/space';
+
+function UnconnectedToggleControl( {
+	__nextHasNoMarginBottom,
+	label,
+	checked,
+	help,
+	className,
+	onChange,
+	disabled,
+}: WordPressComponentProps< ToggleControlProps, 'input', false > ) {
+	function onChangeToggle( event: ChangeEvent< HTMLInputElement > ) {
+		onChange( event.target.checked );
+	}
+	const instanceId = useInstanceId( ToggleControl );
+	const id = `inspector-toggle-control-${ instanceId }`;
+
+	const cx = useCx();
+	const classes = cx(
+		'components-toggle-control',
+		className,
+		! __nextHasNoMarginBottom && css( { marginBottom: space( 3 ) } )
```

Comment:
For back compat, we set a custom value (12px) here instead of simply passing through the `__nextHasNoMarginBottom` to BaseControl. This is because ToggleControl was setting a custom margin-bottom value (12px) that was different from BaseControl's margin-bottom (8px).